### PR TITLE
Update first/last tests to avoid non-determinisim and ordering differences

### PR DESCRIPTION
**Pull request by revans2**
This fixes the majority of #812 but it does not fix all of the issues. I'll try to address them in separate PRs as I debug them further.

### CODE REVIEW

**Comment by revans2** [comment]
build

File: integration_tests/src/main/python/qa_nightly_select_test.py [hunk 1]
```diff
@@ -134,6 +146,8 @@ def idfn(val):
         'spark.rapids.sql.hasNans': 'false',
         'spark.rapids.sql.castStringToFloat.enabled': 'true',
         'spark.rapids.sql.castFloatToString.enabled': 'true',
+        # some of the first/last tests need a single partition to work reliably when run on a large cluster.
+        'spark.sql.shuffle.partitions': '1'
```
Comment (tgravescs) [inline integration_tests/src/main/python/qa_nightly_select_test.py:150]: would it be better to split the first/last test to use different confs so we can test others with multiple partitions?

Comment (revans2) [reply integration_tests/src/main/python/qa_nightly_select_test.py]: Sure simple enough to do.

**Comment by revans2** [comment]
build

**Review by tgravescs — APPROVED** [review]

**Pull request merged**